### ISSUE TITLE
Strange mLab issue.

### ISSUE THREAD

**Issue by daimenworrall** [ParsePlatform/parse-server#1893]
Hi guys,
I have come across a strange issue (not sure if it's only mLab related). I have my parse database hosted with mLab. Everything through the parse API is fine. However, if i insert data from another mongo client the data is lost.
I have tried the same code but on a collection not used by parse and everything looks OK there.

Does parse do anything to collections so only parse can use them?

**Comment by daimenworrall** [comment]
After some investigation is seems like Parse is doing some kind of cleanup on the collections, on documents that don't have all the required information. Can anybody confirm this?

**Comment by bohemima** [comment]
What kind of data did you insert?

**Comment by daimenworrall** [comment]
Something similar to
`{"_id":"qwertyuiop", "value": "Test"}`

My hunch is that, because there was no ACL, Parse removed it.

**Comment by bohemima** [comment]
There's also no objectId. Are they still in the database but simply not shown in the dashboard?

**Comment by drew-gross** [comment]
`_id` is translated to `objectId` by Parse Server. The missing ACL might be an issue, also if `value` is not in the schema for that object, that might also be an issue. Parse Server should be deleting anything unless issued a delete request though. Try inserting an exact copy of an object that is already in the collection, but with a different ID, and see if that works.

**Comment by daimenworrall** [comment]
Just to clarify. I'm inserting directly to the database from a mongo driver (nodejs)

The data is visible in the database for a few seconds via mongo shell then gets removed by Parse. 

My database contains _id and not objectid. 

I fixed the issue by inserting ACL entries for each object. I just remembered I was setting the owner of the object as well as the above information. 

IMO parse should assume the owner of the object only has rights if no ACL is set. It shouldn't delete data. This is fairly important going forward, as users have access to the mongo databases.
